test/unit_test/character_sheet_unit_tests/FocusesUnitTests.java

```diff
@@ -0,0 +1,144 @@
+package unit_test.character_sheet_unit_tests;
+
+import static org.junit.Assert.*;
+
+import java.security.InvalidParameterException;
+import java.util.Collection;
+import java.util.HashSet;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+import rpg_database.character_sheet.Focuses;
+import rpg_database.character_sheet.FocusesSet;
+
+public class FocusesUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetFocusesMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Focuses");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.FOCUS, malformedInput);
+	}
+
+	// functional tests
+
+	@Test
+	public void testSetFocusWithHashSet() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		HashSet<Focuses> expectedFocuses = new HashSet<Focuses>();
+		expectedFocuses.add(Focuses.INTIMIDATION);
+		characterSheet.setData(Fields.FOCUS, new FocusesSet(Focuses.INTIMIDATION));
+		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUS);
+		assertEquals(expectedFocuses, actualFocus);
+	}
+
+	@Test
+	public void testSetFocusWithFocusesSet() {
+		CharacterSheet characterSheet = new CharacterSheet("TestCharacterSheet");
+		FocusesSet expectedFocuses = new FocusesSet(Focuses.ETIQUETTE);
+		characterSheet.setData(Fields.FOCUS, expectedFocuses);
+		FocusesSet actualFocus = characterSheet.getData(Fields.FOCUS);
+		assertEquals(expectedFocuses, actualFocus);
+	}
+

```

Comment:
testSetFocusWithHashSet, testSetFocusWithFocusesSet --> itt igazából mindegy melyikkel teszteled, mindkét set equals metódusa a Set interfészből jön, ergo ugyanarra jó a két teszt. Ez nem biztos hogy elsőre triviális, ezért írom ide, ez gyakorlatilag kódduplikátum.

1. Én hashset-tel azt tesztelném hogy a FocusesSet konstruktora megfelelően viselkedik -e (erre nincs speciel teszt), anélkül, hogy a CharacterSheet objektum egyáltalán szerepelne a tesztben.
2. A FocusesSet-tel meg azt tesztelném amit ezzel a két teszttel te is, hogy a characterSheet-ben jól szuperál -e az objektum.